docs/src/introduction_to_catalyst/introduction_to_catalyst.md

```diff
@@ -306,53 +173,80 @@ k \frac{X^2}{2!} \frac{Y^3}{3!} \\
-\frac{dY}{dt} &=  -3 k X^2 Y^3, &
+\frac{dX}{dt} &= -2 k X^2 Y^3, &
+\frac{dY}{dt} &= -3 k X^2 Y^3, &
 \frac{dZ}{dt} &= k X^2 Y^3.
 \end{align*}
 ```
 
----
-## Notes
-1. For each of the preceding models we converted the `ReactionSystem` to, i.e.,
-   ODEs, jumps, or SDEs, we had two paths for conversion:
+## [Catalyst represents its models symbolically, and these can be generated programmatically](@id introduction_to_catalyst_sym_prog)
+Above we have described the basics of model creation and simulation in Catalyst. Before we close this introduction, however, we would also like to describe that Catalyst represents its model equations symbolically (i.e. storing the actual algebraic expressions). Internally, Catalyst harnesses this in various ways (to e.g. boost simulation speed), however, an experienced user can also take direct advantage of it.
+
+Throughout most of Catalyst's documentation, models are declared using the `@reaction_network` macro. However, to gain better access to internal representations, it is typically better to create the model *programmatically*. Below we will create a model programmatically. In the context of this short tutorial, we will not go into all the details of what is going on (these can, however, be found [here](@ref programmatic_CRN_construction)).
+
+A [Repressilator](@ref ref) is a system with three species ($X$, $Y$, and $Z$), each repressing the production of the subsequent species in the circuit. We begin with declaring the model species and parameters.
+```@example intro_prog_sym
+using Catalyst # hide
+t = default_t()
+@species X(t) Y(t) Z(t)
+@parameters K n d
+nothing # hide
+```
+Next, we create a production and a degradation reaction for each species. Normally, we have to list each reaction. However, since we are building our model programmatically, we can utilise a `for`-loop. Finally, the model is created using the the `ReactionSystem` constructor:
+```@example intro_prog_sym
+rxs = []
+for (sp1, sp2) in zip([X, Y, Z], [Z, X, Y])
+    push!(rxs, Reaction(K/(K + sp2^3), [], [sp1]))
+    push!(rxs, Reaction(d, [sp1], []))
+end
+@named repressilator = ReactionSystem(rxs, t)
+repressilator = complete(repressilator)
+```
+We now have a model, `repressilator`, which can be analysed and simulated just like the models we have created previously:
+```@example intro_prog_sym
+using OrdinaryDiffEq, Plots # hide
+u0 = [X => 0.8, Y => 0.5, Z => 0.5]
+tspan = (0.0, 10.)
+ps = [K => 0.1, n => 3, d => 0.5]
+oprob = ODEProblem(repressilator, u0, tspan, ps)
+sol = solve(oprob, Tsit5())
+plot(sol)
+```
+
+If we consider the code above, we use the species and parameters (e.g. `K` and `X`) to designate values and create our reactions. This is possible because the `@species` and `@parameters` macros create these as [*symbolic variables*](@ref ref). This enables us to form algebraic expressions using them. E.g. here we create an expression describing $X$'s production rate:
+```@example intro_prog_sym
+X_prod = K/(K + Z^3)
```

Comment:
I think this part is quite nice in showing of how the symbolic representation is useful. Originally I used normal hill functions to model the repressilator, but then the print-out of this was just
```
hillr(X, v, K, n)
```
which was kind of underwhelming. This gave a better result (and is also a fair way to model a repressilator)